servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TimeoutPublisher.java

```diff
@@ -228,7 +291,7 @@ private void timerFires() {
                 final long nextTimeoutNs = parent.durationNs - (currentTimeNs - lastStartNS);
                 if (nextTimeoutNs <= 0) { // Timeout!
                     offloadTimeout(new TimeoutException("timeout after " + NANOSECONDS.toMillis(parent.durationNs) +
```

Comment:
Users will appreciate if we use `restartAtOnNext` to enhance this exception message by clarifying if a timeout happened between `onNext` or for the terminal event